pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -108,46 +108,75 @@ func getConsoleTemplate(cr *v1.KieApp) v1.ConsoleTemplate {
 func getServersConfig(cr *v1.KieApp, commonConfig *v1.CommonConfig) ([]v1.ServerTemplate, error) {
 	var servers []v1.ServerTemplate
 	if len(cr.Spec.Objects.Servers) == 0 {
-		cr.Spec.Objects.Servers = []v1.KieServerSet{{Deployments: Pint(constants.DefaultKieDeployments)}}
+		cr.Spec.Objects.Servers = []v1.KieServerSet{{}}
 	}
-	names := map[string]bool{}
+	usedNames := map[string]bool{}
 	for index := range cr.Spec.Objects.Servers {
-		kieID := GenKieName(cr.Name, cr.Spec.Objects.Servers[index].Name)
-		if exists := names[kieID]; exists {
-			return []v1.ServerTemplate{}, fmt.Errorf("duplicate kieserver name %s", kieID)
+		serverSet := &cr.Spec.Objects.Servers[index]
+		if serverSet.Name == "" {
+			if len(cr.Spec.Objects.Servers) == 1 {
+				serverSet.Name = fmt.Sprintf("%v-kieserver", cr.Name)
+			} else {
+				serverSet.Name = fmt.Sprintf("%v-kieserver%v", cr.Name, index)
```

Comment:
This is a great idea and works well for the second nameless serverSet... appends a 0 to the serverSet.Name... but when trying to add a 3rd/4th nameless serverSet it tries to append a 0 again instead of incrementing and results in conflict.

Reply:
likely need 2 indexes? one for the serverSet.Name(s) and one for the deployments of that serverset?

Reply:
This is old / obsolete, right?

Reply:
maybe not something we have to solve now? i can show you the issue later... would be a nice-to-have i guess... more intelligence